internal/metric/noop.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+//go:build !telemetry
```

Comment:
question: What does this do?

Reply:
build will use this file if the `telemetry` build flag isn't used. So:
- noop.go will be used if no `telemetry` build flag.
- otel.go will be used if `telemetry` flag.